wordpress-mobile/WordPress-iOS
# Improve media retry

Fixes #11176 and helps minimize #12353

This PR improves the media code when handling media upload fails and retries.

To test:

On the media library of a site try the following:
 - Upload new media
 - Upload new media and then turn off the network connection in the middle of the update
 - Check if failures are reported correctly
 - Turn the connection on again and retry uploads of media

Using the Aztec editor:

 - Create a post with media
 - While media is uploading turn off connection
 - Check if errors are reported on media
 - Check if you can retry uploads after connection is made

 - Create a new  post with media
 - While media is still uploading press Publish
 - Check if upload continues on Post List
 - Disconnect network in the middle
 - See if error reports correctly
 - Reconnect network and retry post publishing.



Update release notes:
